ci/test/03_test_script.sh

```diff
@@ -144,21 +144,74 @@ if [ "$RUN_CHECK_DEPS" = "true" ]; then
   "${BASE_ROOT_DIR}/contrib/devtools/check-deps.sh" .
 fi
 
+function get_interfaces()
+{
+  ifconfig | awk -F ':| ' '/^[^[:space:]]/ { if (!match($1, /^lo/)) { print $1 } }'
+}
+
+function tcpdump_file_for_interface()
+{
+  echo "/tmp/tcpdump_$1"
+}
+
+function traffic_monitor_begin()
+{
+  for ifname in $(get_interfaces) ; do
+    tcpdump -n -i "$ifname" -w "$(tcpdump_file_for_interface "$ifname")" &
+  done
+}
+
+function traffic_monitor_end()
+{
+  # Stop all tcpdump instances (we want the word splitting if "jobs -p" returns more than one PID).
+  # shellcheck disable=SC2046
+  while kill -SIGTERM $(jobs -p) ; do
+    sleep 1
+  done
```

Comment:
To 1. flush the file and 2. so that the next set of tests can start fresh without leftover `tcpdump` processes. Now 1. is not needed because of `-U`, but I will leave it for 2.

In my experiments `SIGTERM` would cause `tcpdump` to exit gracefully - flush the file and then exit. `SIGKILL` would result in unflushed file.

Yes, in some tasks `kill` gives "permission denied" even if docker is running with `--privileged` or `--cap-add ALL`.

Reply:
Then it would be unclear whether the traffic was generated by unit tests, functional tests or fuzz tests. So I start/stop a separate monitoring for each of those set of tests. I perceived this information useful when investigating. It is already tough that it is not known which test generated the traffic.

Somehow I do not like reading from the "live" file while it is being written. `-U` would help not to miss data, but it might happen that by the time we read it the last packet is still being written and is half-written and the reading `tcpdump` is upset by that.

Reply:
Hmm, right. Then I have no idea why `kill` does not work on some. It is ok as long as it works on at least one.

Reply:
In one of the previous pushes I changed it to use distinct tcpdump files for each test run, so 2. from https://github.com/bitcoin/bitcoin/pull/31349#discussion_r1869187689 is not necessary any more. Removed this snippet.